### DISCUSSION TITLE
Predicting signal after dkifit

### DISCUSSION THREAD

**Discussion by cetisca** [dipy/dipy#2597] [Preprocessing]
Hi!

I have quite a basic and quick question about running dki on diffusion data and then running some predictions on the signal. 

This is the code that I am running (using the advice on the dipy page):
dkimodel = dki.DiffusionKurtosisModel(gtab)
dkifit = dkimodel.fit(data, mask=mask)

Then I am running the command for predicting the signal based:
predicted_signal=dki_prediction(dkifit.model_params, gtab,data[60,60,60,0])

The FA/MD outputs seem sensible in terms of values/appearance, but it it seems like the difference between the predicted signal and the real data is really big quite significant, leading to large SSE.
I ran the dki model with and without Gaussian smoothing of the data. 

Is there anything I'm missing?

Thanks,
Cristiana

**Comment by arokem** [comment]
I am not sure. I assume that `data[60,60,60,0]` is the b=0 signal in the voxel that you are trying to predict for? 

It might help if you plotted the predictions and the real signal. For example a plot of the data (real and predicted) as a function of bvalues could help assess the problem:

```
plt.plot(gtab.bvals, predicted_signal, 'o') 
plt.plot(gtab.bvals, data[60, 60, 60], 'o')
```

Also potentially useful is a scatter plot of predicted vs. measured: 

```
plt.scatter(predicted_signal, data[60, 60, 60], 'o')
```
